attachment_azure/models/ir_attachment.py

```diff
@@ -134,14 +134,13 @@ def _get_azure_container(self, container_name=None):
             container_name = self._get_container_name()
         blob_service_client = self._get_blob_service_client()
         container_client = blob_service_client.get_container_client(container_name)
-        try:
-            # Create the container
-            container_client.create_container()
-        except ResourceExistsError:
-            pass
-        except HttpResponseError as error:
-            _logger.exception("Error during the creation of the Azure container")
-            raise exceptions.UserError(str(error))
+        if not any(x['name'] == container_name for x in blob_service_client.list_containers()):
```

Comment:
`list_containers` has a parameter `name_starts_with`, wouldn't it be more effective to ask only for matching container names?

 https://docs.microsoft.com/en-us/python/api/azure-storage-blob/azure.storage.blob.blobserviceclient?view=azure-python#list-containers-name-starts-with-none--include-metadata-false----kwargs-

Reply:
Have you tried using the method `containter_client.exists()`?